websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,275 @@
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${
+					document.title.split("Category:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("Talk:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Browsing a talk page:";
+				presenceData.state = `${
+					document.title.split("Talk:")[1].split(" - Sonic Retro")[0]
+				}`;
+			} else if (document.title.startsWith("Sonic Retro:")) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${
+					document.title.split("Sonic Retro:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View Article",
+						url: location.href
+					}
+				];
+			} else if (document.title.startsWith("User:")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = "Viewing a user's page:";
+				presenceData.state = `@${
+					document.title.split("User:")[1].split(" - Sonic Retro")[0]
+				}`;
+				presenceData.buttons = [
+					{
+						label: "View User",
+						url: location.href
+					}
+				];
+			} else if (
+				pathname.includes("/File:" || "/Sonic_Retro:General_disclaimer")
+			) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (
+				(pathname === "/index.php" && document.title.includes("Search")) ||
+				pathname.includes("/Special:Search") ||
+				pathname.includes("/Special:Search#")
+			) {
+				presenceData.largeImageKey = "search";
+				if (document.title.startsWith("Search -"))
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for ${
+						document.title
+							.split("Search results for ")[1]
+							.split(" - Sonic Retro")[0]
+					}`;
+				}
+			} else if (pathname === "/index.php") {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (pathname.startsWith("User_talk:")) {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article:";
+				presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case "forums.sonicretro.org": {
+			presenceData.smallImageKey = "forums_512";
+			presenceData.smallImageText = "forums.sonicretro.org";
+			if (search) {
+				const forumTitle = document.title.split(
+					" | Sonic and Sega Retro Forums"
+				)[0];
+				if (search.startsWith("?watched/fourums")) {
```

Comment:
you sure this works? Can you show me a URL with this?

Reply:
I misspelled forums 😤